atom/browser/extensions/atom_extension_host_delegate.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/extensions/atom_extension_host_delegate.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "atom/browser/extensions/atom_extension_web_contents_observer.h"
+#include "base/lazy_instance.h"
+#include "base/logging.h"
+#include "extensions/browser/media_capture_util.h"
+#include "extensions/browser/serial_extension_host_queue.h"
+
+namespace extensions {
+
+AtomExtensionHostDelegate::AtomExtensionHostDelegate() {}
+
+AtomExtensionHostDelegate::~AtomExtensionHostDelegate() {}
+
+void AtomExtensionHostDelegate::OnExtensionHostCreated(
+    content::WebContents* web_contents) {
+  AtomExtensionWebContentsObserver::CreateForWebContents(web_contents);
+}
+
+void AtomExtensionHostDelegate::OnRenderViewCreatedForBackgroundPage(
+    ExtensionHost* host) {}
+
+content::JavaScriptDialogManager*
+AtomExtensionHostDelegate::GetJavaScriptDialogManager() {
+  // TODO(jamescook): Create a JavaScriptDialogManager or reuse the one from
+  // content_shell.
+  NOTREACHED();
+  return NULL;
+}
+
+void AtomExtensionHostDelegate::CreateTab(
+    std::unique_ptr<content::WebContents> web_contents,
+    const std::string& extension_id,
+    WindowOpenDisposition disposition,
+    const gfx::Rect& initial_rect,
+    bool user_gesture) {
+  // TODO(jamescook): Should app_shell support opening popup windows?
+  NOTREACHED();
+}
+
+void AtomExtensionHostDelegate::ProcessMediaAccessRequest(
+    content::WebContents* web_contents,
+    const content::MediaStreamRequest& request,
+    content::MediaResponseCallback callback,
+    const Extension* extension) {
+  // Allow access to the microphone and/or camera.
+  media_capture_util::GrantMediaStreamRequest(web_contents, request,
```

Comment:
Not a blocker to implement for this PR, but all of these methods need to be routed to existing Electron APIs -- for this PR, maybe adding a `TODO` so that we remember all the things that need to be done